pkg/utils/errors.go

```diff
@@ -1,9 +1,23 @@
 package utils
 
-import "errors"
+import (
+	"errors"
+	"strings"
+)
 
 var (
 	ErrNotFound                   = errors.New("resource was not found")
 	ErrInsufficientCidrBlocks     = errors.New("InsufficientCidrBlocks: The specified subnet does not have enough free cidr blocks to satisfy the request")
 	ErrMsgProviderAndPoolNotFound = "cannot find the instance provider and pool from the cache"
+	NotRetryErrors                = []string{ErrInsufficientCidrBlocks.Error()}
 )
+
+// ShouldRetryOnError returns true if the error is retryable, else returns false
+func ShouldRetryOnError(err error) bool {
+	for _, e := range NotRetryErrors {
+		if strings.HasPrefix(err.Error(), e) {
```

Comment:
How confident this prefix will be matched for a long prefix `InsufficientCidrBlocks: The specified subnet does not have enough free cidr blocks to satisfy the request`?

Reply:
That's a good point. I will make it match the shorter string `InsufficientCidrBlocks`.